features/bootstrap/SwaggerContext.php

```diff
@@ -0,0 +1,275 @@
+    }
+
+    /**
+     * @Then the value of the node ":node" of the operation ":operation" of the Swagger class ":class" is ":value"
+     */
+    public function assertOperationNodeValueIs($nodeName, $operationMethod, $className, $value)
+    {
+        $property = $this->getOperation($operationMethod, $className);
+
+        \PHPUnit_Framework_Assert::assertEquals($this->propertyAccessor->getValue($property, $nodeName), $value);
+    }
+
+    /**
+     * @Then :nb operations are available for Swagger class ":class"
+     */
+    public function assertNbOperationsExist($nb, $className)
+    {
+        $operations = $this->getOperations($className);
+
+        \PHPUnit_Framework_Assert::assertEquals($nb, count($operations));
+    }
+
+    /**
+     * @Then :nb properties are available for Swagger class ":class"
+     */
+    public function assertNbPropertiesExist($nb, $className)
+    {
+        $properties = $this->getProperties($className);
+
+        \PHPUnit_Framework_Assert::assertEquals($nb, count($properties));
+    }
+
+    /**
+     * @Then ":prop" property doesn't exist for the Swagger class ":class"
+     */
+    public function assertPropertyNotExist($propertyName, $className)
+    {
+        try {
+            $this->getProperty($propertyName, $className);
+
+            throw new \PHPUnit_Framework_ExpectationFailedException(
+                sprintf('The property "%s" for the class "%s" exist.', $propertyName, $className)
+            );
+        } catch (\Exception $exception) {
+            // an exception must be catched
+        }
+    }
+
+    /**
+     * @Then ":prop" property is required for Swagger class ":class"
+     */
+    public function assertPropertyIsRequired(string $propertyName, string $className)
+    {
+        $classInfo = $this->getClassInfos($className);
+        if (!in_array($propertyName, $classInfo->{'required'})) {
+            throw new \Exception(sprintf('Property "%s" of class "%s" is not required', $propertyName, $className));
+        }
+    }
+
+    /**
+     * @param string $propertyName
+     * @param string $className
+     *
+     * @throws Exception
+     *
+     * @return array
+     */
+    private function getProperty($propertyName, $className)
+    {
+        $properties = $this->getProperties($className);
+        $propertyInfos = null;
+        foreach ($properties as $classPropertyName => $property) {
+            if ($classPropertyName === $propertyName) {
+                $propertyInfos = $property;
+            }
+        }
+
+        if (empty($propertyInfos)) {
+            throw new \Exception(sprintf('Property "%s" of class "%s" does\'nt exist', $propertyName, $className));
+        }
+
+        return $propertyInfos;
+    }
+
+    /**
+     * Gets an operation by its method name.
+     *
+     * @param string $className
+     * @param string $method
+     *
+     * @throws Exception
+     *
+     * @return array
+     */
+    private function getOperation($method, $className)
```

Comment:
as we are in php7 we can force the typehint everywhere and remove useless phpdoc